dev/io.openliberty.netty.internal.tls.impl/src/io/openliberty/netty/internal/tls/impl/NettyTlsProviderImpl.java

```
@@ -0,0 +1,327 @@
+     * @param alias
+     * @param properties
+     * @param isInbound
+     * @param host
+     * @param port
+     * @param endPoint
+     * @param isZWebContainerChain
+     * @return
+     * @throws Exception
+     */
+    private static SSLContext getSSLContext(String alias, Properties properties, boolean isInbound, String host,
+            String port, String endPoint, Boolean isZWebContainerChain) throws Exception {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "host=" + host + " port=" + port + " endPoint=" + endPoint);
+        }
+
+        // Set up the parameters needed to call into the JSSEHelper to extract official
+        // SSL Props.
+        String direction = (isInbound) ? Constants.DIRECTION_INBOUND : Constants.DIRECTION_OUTBOUND;
+        final Map<String, Object> connectionInfo = new HashMap<String, Object>();
+        connectionInfo.put(Constants.CONNECTION_INFO_DIRECTION, direction);
+        connectionInfo.put(Constants.CONNECTION_INFO_REMOTE_HOST, host);
+        connectionInfo.put(Constants.CONNECTION_INFO_REMOTE_PORT, port);
+        connectionInfo.put(Constants.CONNECTION_INFO_ENDPOINT_NAME, endPoint);
+
+        // PK40641 - handle channel props that do not need JSSEHelper
+        Properties props = null;
+        boolean useJSSEHelper = (null != alias);
+        if (!useJSSEHelper) {
+            // 436920 - check for ssl properties put programmatically on the thread.
+            // Calling into JSSEHelper.getProperties() does this same action
+            try {
+                props = AccessController.doPrivileged(new PrivilegedExceptionAction<Properties>() {
+                    @Override
+                    public Properties run() throws Exception {
+                        return com.ibm.websphere.ssl.JSSEHelper.getInstance().getSSLPropertiesOnThread();
+                    }
+                });
+            } catch (Exception e) {
+                // no ffdc required
```

Seems unusual to swallow such a wide exception?

---

It does seem a bit unusual. The TLS implementation was taken and adapted from the original channel code in SSLChannel [here](https://github.com/OpenLiberty/open-liberty/blob/2fd4a880754c37a988c5ed9ac4f1ea5988e465d6/dev/com.ibm.ws.channel.ssl/src/com/ibm/ws/channel/ssl/internal/SSLChannel.java#L372) from what I can deduce in the code I believe it should be fine and that it is expected in some occasions

---

OK